File: python-service/app/routers/health.py
# app/routers/health.py

from fastapi import APIRouter
from datetime import datetime, timezone
import socket

router = APIRouter()

@router.get("/health")
def health_check():
    """
    Health check endpoint for service monitoring.

    Returns:
        JSON object including service name, host machine, and UTC timestamp.
    """
    return {
        "status": "ok",
        "service": "python-mail",
        "host": socket.gethostname(),
        "time": datetime.now(timezone.utc).isoformat()
    }

File: frontend/src/types/health.ts
/**
 * Response type for /health endpoints (backend & python)
 *
 * Example:
 * {
 *   "status": "ok",
 *   "service": "backend",
 *   "timestamp": "2025-09-12T14:30:00Z"
 * }
 */

export type HealthResponse = {
  status: "ok" | "healthy" | string;

  service?: string;

  timestamp?: string;
};
